src/init.cpp

```diff
@@ -925,21 +925,17 @@ bool AppInitParameterInteraction(const ArgsManager& args, bool use_syscall_sandb
     const int nBind = std::max(nUserBind, size_t(1));
     nUserMaxConnections = args.GetIntArg("-maxconnections", DEFAULT_MAX_PEER_CONNECTIONS);
     nMaxConnections = std::max(nUserMaxConnections, 0);
-    const int nFDMin = nBind + RESERVED_FILE_DESCRIPTORS;
+    const int nRpcThreads = std::max<int>(args.GetIntArg("-rpcthreads", DEFAULT_HTTP_THREADS), 1L);
+    const int nFDMin = nBind + nRpcThreads + RESERVED_FILE_DESCRIPTORS;
 
     nFD = RaiseFileDescriptorLimit(nMaxConnections + nFDMin);
+    if (nFD < nFDMin) {
+        return InitError(strprintf(_("Not enough file descriptors available. %d available, %d required."), nFD, nFDMin));
+    }
 
-#ifdef USE_POLL
-    int fd_max = nFD;
-#else
-    int fd_max = FD_SETSIZE;
```

Comment:
Needs an explanation of why it is ok to take `FD_SETSIZE` completely out of the equation in the commit description.